### DISCUSSION TITLE
ref.listen with asyncvalue

### DISCUSSION THREAD

**Discussion by angelru** [rrousselGit/riverpod#2946] [Q&A] [answered]
Hello, I'm new to riverpod

I'm using this code in my widget to display a snackbar when some error occurs in my asynchronous call:

```
    ref.listen<AsyncValue>(authNotifierProvider, (_, state) {
      showSnackbar(context, state.error.toString());
    });
    final state = ref.watch(authNotifierProvider);
```

If you press the button and make the call, it shows the error, but if I press it again the snackbar is shown twice, why does this happen? Is it because the state changes from true to false and the error is logged? What is the way to deal with it?

**Comment by rrousselGit** [comment]
Check state.isLoading

**Reply by angelru** [reply]
where do I check it? in the snackbar?

**Reply by rrousselGit** [reply]
In the listen callback

**Reply by angelru** [reply]
```
    ref.listen<AsyncValue>(authNotifierProvider, (_, state) {
if (state.isLoading)
      showSnackbar(context, state.error.toString());
    });
```
its ok??

**Reply by rrousselGit** [reply, marked as answer]
You want !state.isLoading

**Reply by angelru** [reply]
```
  ref.listen(authNotifierProvider, (_, state) {
      if (!state.isLoading && state.hasError) {
        showSnackbar(context, context.loc.loginError);
      }
    });

```

**Comment by LScodeblack** [comment]
ref.listen(authControllerProvider, (_, state) {
      if (!state.isLoading && state.hasError) {
        ScaffoldMessenger.of(context).showSnackBar(
          SnackBar(
              backgroundColor: Colors.red,
              content: ErrorText(error: state.error)),
        );
      }
    });
    
    this still shows snackbar twice.

**Reply by NguyenKhaiHoan** [reply]
I same you issue, do you fix this?
